CHANGELOG/changelog

```diff
@@ -0,0 +1,329 @@
+  end
+})
+
+local function table_keys(t)
+  if type(t) ~= "table" then
+    return t
+  end
+  local keys = {}
+  for k, _ in pairs(t) do
+    table.insert(keys, k)
+  end
+  return keys
+end
+
+local function parse_github_ref(reference_type, references)
+  if references == nil or references == lyaml.null then
+    return nil
+  end
+  local parsed_references = {}
+  for i, ref in ipairs(references or {}) do
+    local repo = GITHUB_KONG_REPO
+    local ref_no = tonumber(ref)
+    local name = "#" .. ref
+    if not ref_no then -- pr is a/b:1 format
+      local parts = pl_stringx.split(ref, ":")
+      repo = parts[1]
+      ref_no = parts[2]
+      name = pl_stringx.replace(tostring(ref), ":", " #")
+    end
+    parsed_references[i] = {
+      id = ref_no,
+      name = name,
+      link = string.format(GITHUB_REFERENCE[reference_type], repo, ref_no),
+    }
+  end
+  return parsed_references
+end
+
+
+local function parse_jiras(jiras)
+  local jira_items = {}
+  for i, jira in ipairs(jiras or {}) do
+    jiras[i] = {
+      id = jira,
+      link = JIRA_BASE_URL .. jira
+    }
+  end
+  return jira_items
+end
+
+
+local function is_yaml(filename)
+  return pl_stringx.endswith(filename, ".yaml") or
+    pl_stringx.endswith(filename, ".yml")
+end
+
+local function is_empty_table(t)
+  if next(t) == nil then
+    return true
+  end
+  return false
+end
+
+local function compile_template(data, template)
+  local compile_env = {
+    _escape = ">",
+    _brackets = "{}",
+    _debug = true,
+    pairs = pairs,
+    ipairs = ipairs,
+    tostring = tostring,
+    is_empty_table = is_empty_table,
+  }
+
+  compile_env = pl_tablex.merge(compile_env, data, true) -- union
+  local content, err = pl_template.substitute(template, compile_env)
+  if not content then
+    return nil, "failed to compile template: " .. err
+  end
+
+  return content
+end
+
+local function absolute_path(...)
+  local path = CHANGELOG_PATH
+  for _, p in ipairs({...}) do
+    path = pl_path.join(path, p)
+  end
+  return path
+end
+
+local function collect_files(folder)
+  local files
+  if pl_path.exists(folder) then
+    files = assert(pl_dir.getfiles(folder))
+    if files then
+      table.sort(files)
+    end
+  end
+  local sorted_files = {}
+  for _, filename in ipairs(files or {}) do
+    if is_yaml(filename) then
+      table.insert(sorted_files, filename)
+    end
+  end
+
+  return sorted_files
+end
+
+
+local function collect_folder(folder)
+  local data = {
+    features = {},
+    bugfixes = {},
+    breaking_changes = {},
+    dependencies = {},
+    deprecations = {},
+  }
+
+  local map = {
+    feature = "features",
+    bugfix = "bugfixes",
+    breaking_change = "breaking_changes",
+    dependency = "dependencies",
+    deprecation = "deprecations",
+  }
+
+  local files = collect_files(folder)
+  for _, filename in ipairs(files) do
+    local content = assert(pl_file.read(filename))
+    local entry = assert(lyaml.load(content))
+    -- validate entry?
```

Comment:
Yes please.

Reply:
All changelog files of the master branch have been validated by the CI `Validate changelog` job.